crates/factor-outbound-http/src/lib.rs

```diff
@@ -38,19 +51,90 @@ impl Factor for OutboundHttpFactor {
         _ctx: PrepareContext<Self>,
         builders: &mut InstanceBuilders<T>,
     ) -> anyhow::Result<Self::InstanceBuilder> {
-        let allowed_hosts = builders
-            .get_mut::<OutboundNetworkingFactor>()?
-            .allowed_hosts();
+        let outbound_networking = builders.get_mut::<OutboundNetworkingFactor>()?;
+        let allowed_hosts = outbound_networking.allowed_hosts();
+        let component_tls_configs = outbound_networking.component_tls_configs().clone();
         Ok(InstanceState {
-            allowed_hosts,
             wasi_http_ctx: WasiHttpCtx::new(),
+            allowed_hosts,
+            component_tls_configs,
+            request_interceptor: None,
         })
     }
 }
 
 pub struct InstanceState {
-    allowed_hosts: OutboundAllowedHosts,
     wasi_http_ctx: WasiHttpCtx,
+    allowed_hosts: OutboundAllowedHosts,
+    component_tls_configs: ComponentTlsConfigs,
+    request_interceptor: Option<Box<dyn OutboundHttpInterceptor>>,
+}
+
+impl InstanceState {
+    pub fn set_request_interceptor(
+        &mut self,
+        interceptor: impl OutboundHttpInterceptor + 'static,
+    ) -> anyhow::Result<()> {
+        if self.request_interceptor.is_some() {
+            anyhow::bail!("set_request_interceptor can only be called once");
+        }
```

Comment:
Are we sure we don't want to just silently overwrite the interceptor?

Reply:
Pretty sure. Since this is used with dependency inversion the end developer could be surprised when composing two factors that call this if one of them silently loses its interceptor.

The real solution would be to allow multiple interceptors but that is a little complicated in rust so I punted.